plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryListerModule.java

```diff
@@ -38,12 +37,10 @@ public void configure(Binder binder)
         if (cachingDirectoryLister.isPresent()) {
             CachingDirectoryLister directoryLister = cachingDirectoryLister.get();
             binder.bind(DirectoryLister.class).toInstance(directoryLister);
-            binder.bind(TableInvalidationCallback.class).toInstance(directoryLister);
         }
         else {
             binder.bind(CachingDirectoryLister.class).in(Scopes.SINGLETON);
             binder.bind(DirectoryLister.class).to(CachingDirectoryLister.class);
-            binder.bind(TableInvalidationCallback.class).to(CachingDirectoryLister.class);
```

Comment:
change the above to `binder.bind(DirectoryLister.class).to(CachingDirectoryLister.class).in(Scopes.SINGLETON);`